modules/builtin/src/backend/variables/complex.ts

```diff
@@ -0,0 +1,53 @@
+import { BoxedVariable, PrimitiveVarType } from 'botpress/sdk'
+import { BaseVariable } from 'common/variables'
+
+import common from './common'
+
+class BoxedComplex extends BaseVariable<string> {
+  constructor(args) {
+    super(args)
+  }
+
+  trySet(value: string, confidence: number) {
```

Comment:
As I think about it, we should leave the extraction job to the NLU and don't validate the value here.  That's because a complex type could potentially be anything (because of examples, such as names etc)